src/localized-text/LocalizedAppearanceEditor.component.js

```diff
@@ -18,7 +35,29 @@ const styles = {
     field: {
         width: '100%',
     },
+    loaderWrap: {
+        textAlign: 'center',
+        padding: 12,
+    },
+    error: {
+        color: 'red',
+        padding: 12,
+    }
 };
+const SYSTEM_DEFAULT = '@@__SYSTEM_DEFAULT__@@';
+
+const LOCALIZED_SETTING_KEYS = [
+    'applicationTitle',
+    'keyApplicationIntro',
+    'keyApplicationNotification',
+    'keyApplicationFooter',
+    'keyApplicationRightFooter',
+];
+
+const readDefaultAppearanceSettingsFromStore = () => 
```

Comment:
So I see the logic in getAppearanceSettings depends on this, and you're checking for settingsStore.state. I assume then that it's possible for settingsStore to not have a state prop?

If so, I'd first do a check if settingsStore has a state prop at the beginning of readDefaultAppearanceSettingsFromStore, and return a rejected promise if it doesn't. That way the catch in getAppearanceSettings can deal with it. Otherwise the promise in readDefaultAppearanceSettingsFromStore will resolve with an array filled with undefined's

Reply:
`settingsStore` is initialized when the app loads, so it will have a state. I don't think this comment is relevant...

Calling `settingsActions.load` when app is initialised:
https://github.com/dhis2/settings-app/blob/6fed97688572d7e5ac41baf0619749c0658f412e/src/settings-app.js#L173

Initialising the store in `settingsActions.load`
https://github.com/dhis2/settings-app/blob/bcf7dd047d346f1d2cebd4d697718d4b2ba1b8ff/src/settingsActions.js#L101

Reply:
> settingsStore is initialized when the app loads, so it will have a state. I don't think this comment is relevant...

I'm basing this on the check you're doing in line 59. I assumed you were doing the check because it's possible for the prop to not be there. Otherwise why check?

Reply:
Sorry I managed to misinterpret both of your comments... Not very good at context-switching....
Your remark was relevant and I have changed my code a bit.